ihatemoney/models.py

```diff
@@ -320,6 +322,42 @@ def switch_currency(self, new_currency):
         db.session.add(self)
         db.session.commit()
 
+    def import_bills(self, bills: list):
+        """Import bills from a list of dictionaries"""
+        # Add members not already in the project
+        members_project = [str(m) for m in self.members]
+        members_new = [
+            m for m in get_members(bills) if str(m[0]) not in members_project
+        ]
+        for m in members_new:
+            Person(name=m[0], project=self, weight=m[1])
+        db.session.commit()
+
+        # Import bills not already in the project
+        bills_project = self.get_pretty_bills()
+        id_dict = {m.name: m.id for m in self.members}
+        for b in bills:
+            same = False
+            for b_p in bills_project:
+                if same_bill(b_p, b):
+                    same = True
+                    break
+            if not same:
+                # Create bills
+                db.session.add(
+                    Bill(
+                        amount=b["amount"],
```

Comment:
We probably should check that the data contains this key before using it, otherwise it may fail.

Reply:
Or we might want to try/catch this entire block and display an error message with the format of the required data.

Reply:
import_bills is already called from a try/catch block in web.import_project
but I agree we should add another one in import_bills as a safeguard in case we use the method elsewhere
done !